kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h

```diff
@@ -1459,24 +1322,25 @@ class ResidualBasedEliminationBuilderAndSolver
         }
     }
 
-    //**************************************************************************
-
-    void AssembleLHS_CompleteOnFreeRows(
-        TSystemMatrixType& A,
-        LocalSystemMatrixType& LHS_Contribution,
-        Element::EquationIdVectorType& EquationId
-    )
+    /**
+     * @brief This method assembles the LHS of the system (on free rows)
+     * @param rA The LHS to assemble
+     * @param rLHSContribution The local LHS contribution
+     * @param rEquationId The equation id
+     */
+    void AssembleLHSCompleteOnFreeRows(
+        TSystemMatrixType& rA,
+        LocalSystemMatrixType& rLHSContribution,
+        EquationIdVectorType& rEquationId
+        )
     {
-        unsigned int local_size = LHS_Contribution.size1();
-        for (unsigned int i_local = 0; i_local < local_size; i_local++)
-        {
-            unsigned int i_global = EquationId[i_local];
-            if (i_global < BaseType::mEquationSystemSize)
-            {
-                for (unsigned int j_local = 0; j_local < local_size; j_local++)
-                {
-                    int j_global = EquationId[j_local];
-                    A(i_global, j_global) += LHS_Contribution(i_local, j_local);
+        const SizeType local_size = rLHSContribution.size1();
+        for (IndexType i_local = 0; i_local < local_size; ++i_local) {
+            const IndexType i_global = rEquationId[i_local];
+            if (i_global < BaseType::mEquationSystemSize) {
+                for (IndexType j_local = 0; j_local < local_size; ++j_local) {
+                    const IndexType j_global = rEquationId[j_local];
+                    rA(i_global, j_global) += rLHSContribution(i_local, j_local);

```

Comment:
no ward? (not threadsafe)

Reply:
Gasp, probably you found just for coincidence the reason of the fail on the RVE test

Reply:
is this solved?

Reply:
ping